### PR TITLE
feat: Add ignorePattern option.

### LINKED ISSUE #2832: Consider adding --ignore-pattern option
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Add support for `--ignore-pattern` in the CLI, similar to the one from eslint:

```
  --ignore-pattern [String]    Pattern of files to ignore (in addition to those in .eslintignore)
```

> Which rule, if any, is this issue related to?

None

> What CSS is needed to reproduce this issue?

None

> What stylelint configuration is needed to reproduce this issue?

None

> Which version of stylelint are you using?

`7.13.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No

> What did you expect to happen?

Doesn't apply.

> What actually happened (e.g. what warnings or errors you are getting)?

Doesn't apply.

### PR DESCRIPTION
**Pull request by satazor**
Closes #2832

### CODE REVIEW

File: lib/standalone.js [hunk 1]
```diff
@@ -117,22 +118,27 @@ module.exports = function(
         // Check for file existence
         return new Promise((resolve, reject) => {
           if (!absoluteCodeFilename) {
-            reject()
```
Comment (satazor) [inline lib/standalone.js]: These changes are unrelated. I guess prettier/eslint made those.

**Comment by satazor** [comment]
@evilebottnawi Thanks! Hoping we can get this quickly :D

**Comment by satazor** [comment]
This PR is quite simple and I really need this feature. Could you guys review this quicker? Thanks

**Comment by ntwb** [comment]
@satazor Apologies, I'm trying to sort out what and why Prettier is making the extra changes in this PR

**Comment by hudochenkov** [comment]
@ntwb maybe because our code initially was formatted with Prettier older than 1.5.3, which @satazor might use. Even minor version might change formatting. I think we should use `~` in `package.json` for Prettier as we do with linters.

**Comment by satazor** [comment]
I've committed with `-n` to bypass prettier so that it doesn't get into the way.

**Comment by satazor** [comment]
Bump.

**Comment by jeddy3** [comment]
@satazor Thanks for this. LGTM. 

Let's try to get this into [`8.1.0`](https://github.com/stylelint/stylelint/issues/2838)

> Could you guys review this quicker? Thanks

A couple of the core team have been away (holidays, relocating etc) for the last few weeks. And so, we're currently working through a backlog of issues.

**Comment by ntwb** [comment]
Changelog updated via [14163e0](https://github.com/stylelint/stylelint/commit/14163e0373b13c95bf7a5f2879ae4d392f4a3085?short_path=4ac32a7#diff-4ac32a78649ca5bdd8e0ba38b7006a1e)
> -   Added: `ignorePattern` option (`--ignore-pattern` in CLI), to allow patterns of files to ignored (#2834).
